### PR TITLE
Feature/685 energy split

### PR DESCRIPTION
**Pull request by bartpleiter**
This pull request introduces a new DAZLs-based component splitter for energy data analysis, adds comprehensive unit tests for both the DAZLs and constant component splitters, and ensures all new files are properly licensed. The DAZLs splitter uses a pre-trained linear model to decompose total energy consumption into wind, solar, and other components, and includes robust error handling and configuration options.

**New DAZLs component splitter:**

* Added `DazlsComponentSplitter` and `DazlsComponentSplitterConfig` in `dazls_component_splitter.py`, implementing a splitter that uses a pre-trained DAZLs model to split energy load into wind, solar, and other components, with validation of required input features and robust error handling.
* Included the corresponding model license file for the pre-trained DAZLs model artifact.

**Unit tests for component splitters:**

* Added comprehensive unit tests for `DazlsComponentSplitter`, covering prediction output, input feature creation, and component validation.
* Added thorough unit tests for `ConstantComponentSplitter`, including configuration validation, prediction correctness, and utility methods for known solar and wind configurations.
* Added missing license headers to the test module’s `__init__.py`.

### CODE REVIEW

File: packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    Example:
+        Basic usage:
+
+        >>> from openstef_core.types import EnergyComponentType
+        >>> config = DazlsComponentSplitterConfig(
+        ...     source_column="total_load",
+        ...     components=[EnergyComponentType.SOLAR, EnergyComponentType.WIND, EnergyComponentType.OTHER],
+        ... )
+        >>> splitter = DazlsComponentSplitter(config)
+        >>> # components = splitter.predict(time_series_data)
```
Comment (MvLieshout) [inline packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py]: I guess this is commented out to avoid docstring testing? I think you could also use 
```suggestion
        >>> components = splitter.predict(time_series_data) # doctest: +SKIP
```
This way, we can easily see later where we should still fix the doctests.

File: packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            {
+                "radiation": df[radiation_col],
+                "windspeed_100m": df[wind_col],
+                "total_load": df[source_col],
+            },
+            index=df.index,
+        )
+
+        # Drop rows with NaN values
+        return input_df.dropna()  # pyright: ignore[reportUnknownMemberType]
```
Comment (MvLieshout) [inline packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py]: Do we drop the empty columns earlier? If not, this could return an empty df right? Do we want to check for that?

File: packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+            # Calculate "other" component as residual
+            forecasts[EnergyComponentType.OTHER] = (
+                input_df["total_load"] - forecasts[EnergyComponentType.SOLAR] - forecasts[EnergyComponentType.WIND]
+            )
+
+            # Reindex to match original input, fill missing with 0
+            components_df = forecasts.reindex(index=data.data.index, fill_value=0.0)
+
+        except (ValueError, KeyError, AttributeError) as e:
```
Comment (MvLieshout) [inline packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py]: I guess this is how we do this in OpenSTEFv3 as well? Why not raise error? Returning zero's feels like failing silently.

File: packages/openstef-models/tests/unit/models/component_splitting/test_dazls_component_splitter.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+def sample_splitter_config() -> DazlsComponentSplitterConfig:
+    """Create sample DAZLs component splitter configuration."""
+    return DazlsComponentSplitterConfig(
+        source_column="load",
+        radiation_column="radiation",
+        windspeed_100m_column="windspeed_100m",
+    )
+
+
+def test_dazls_component_splitter__predict_returns_correct_components(
```
Comment (MvLieshout) [inline packages/openstef-models/tests/unit/models/component_splitting/test_dazls_component_splitter.py]: Might be nice to also test the errored case (where the values are zero now).

File: packages/openstef-models/tests/unit/models/component_splitting/test_linear_component_splitter.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    Returns:
+        TimeSeriesDataset with load, radiation, and windspeed values spanning 24 hours.
+    """
+    data = pd.DataFrame(
+        {
+            "load": [100.0, 150.0, 200.0, 250.0, 300.0],
+            "radiation": [0.0, 0.0, 10.0, 50.0, 100.0],
+            "windspeed_100m": [10.0, 5.0, 0.0, 3.0, 7.0],
+        },
+        index=pd.date_range(datetime.fromisoformat("2025-01-01T00:00:00"), periods=5, freq="1h"),
```
[the remark this answers is not in the document]
Comment (bartpleiter) [reply packages/openstef-models/tests/unit/models/component_splitting/test_linear_component_splitter.py:31]: No, we also do not do this in other files

File: packages/openstef-models/tests/unit/models/component_splitting/test_constant_component_splitter.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+@pytest.fixture
+def sample_timeseries_dataset() -> TimeSeriesDataset:
+    """Create sample time series data with typical energy load values.
+
+    Returns:
+        TimeSeriesDataset with load values spanning 4 hours for component splitting tests.
+    """
+    data = pd.DataFrame(
+        {"load": [100.0, 200.0, 300.0, 400.0]},
+        index=pd.date_range(datetime.fromisoformat("2025-01-01T00:00:00"), periods=4, freq="1h"),
```
[the remark this answers is not in the document]
Comment (bartpleiter) [reply packages/openstef-models/tests/unit/models/component_splitting/test_constant_component_splitter.py:27]: No, we also do not do this in other files

File: packages/openstef-models/tests/unit/models/component_splitting/test_constant_component_splitter.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+    # Assert - Configuration is correct for solar park
+    assert splitter.is_fitted
+    assert splitter.config.source_column == "load"
+    assert splitter.config.component_ratios == {EnergyComponentType.SOLAR: 1.0}
+
+    # Test prediction
+    data = pd.DataFrame(
+        {"load": [100.0, 200.0, 300.0]},
+        index=pd.date_range(datetime.fromisoformat("2025-01-01T00:00:00"), periods=3, freq="1h"),
```
[the remark this answers is not in the document]
Comment (bartpleiter) [reply packages/openstef-models/tests/unit/models/component_splitting/test_constant_component_splitter.py:116]: No, we also do not do this in other files

File: packages/openstef-models/tests/unit/models/component_splitting/test_constant_component_splitter.py [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+    # Assert - Configuration is correct for wind farm
+    assert splitter.is_fitted
+    assert splitter.config.source_column == "load"
+    assert splitter.config.component_ratios == {EnergyComponentType.WIND: 1.0}
+
+    # Test prediction
+    data = pd.DataFrame(
+        {"load": [100.0, 200.0, 300.0]},
+        index=pd.date_range(datetime.fromisoformat("2025-01-01T00:00:00"), periods=3, freq="1h"),
```
[the remark this answers is not in the document]
Comment (bartpleiter) [reply packages/openstef-models/tests/unit/models/component_splitting/test_constant_component_splitter.py:138]: No, we also do not do this in other files

File: packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+
+    def __init__(self, config: DazlsComponentSplitterConfig) -> None:
+        """Initialize the DAZLs component splitter.
+
+        Args:
+            config: Configuration with model path and column names.
+        """
+        super().__init__()
+        self._config = config
+        self._model = joblib.load(self.config.dazls_model_path)  # type: ignore[reportUnknownMemberType]
```
[the remark this answers is not in the document]
Comment (bartpleiter) [reply packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py]: I think this is not needed given the tiny size of the model and the way the component splitter is being used

File: packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py (hunk 1 above)
Comment (bartpleiter) [reply packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py]: Cool! did not know that. I'll also add that to the existing constant component splitter

File: packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py (hunk 2 above)
Comment (bartpleiter) [reply packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py]: I think you are right, I added a check for it.

File: packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py (hunk 3 above)
Comment (bartpleiter) [reply packages/openstef-models/src/openstef_models/models/component_splitting/dazls_component_splitter.py]: yeah good point. I copied this logic from openstef 3, agree that the caller should catch and decide on the fallback

File: packages/openstef-models/tests/unit/models/component_splitting/test_dazls_component_splitter.py (hunk 4 above)
Comment (bartpleiter) [reply packages/openstef-models/tests/unit/models/component_splitting/test_dazls_component_splitter.py]: Now that I removed the error handling fallback state, I think this is not needed anymore
